app/controllers/articles_controller.rb

```diff
@@ -18,7 +20,28 @@ def show
     end
 
     def index
-        @article = Article.all
+    
+
+        @last_page = (Article.all.count / ARTICLES_PER_PAGE).floor

```

Comment:
I could be wrong here but I think you won't be able to see your last page if there is an amount of articles that isn't evenly divisible by ARTICLE_PER_PAGE. example: 11 (total articles) / 10 (ARTICLES_PER_PAGE) = (1.1).floor = 1 You wont see that left over article.